From 3a84081cbd982ce1bd9456eca5b1b03cd495e0fe Mon Sep 17 00:00:00 2001
From: James Almer <jamrial@gmail.com>
Date: Sun, 10 Nov 2019 22:15:44 -0300
Subject: [PATCH] avcodec/librav1e: free the RaPacket after using it

Fixes leaks.

Reviewed-by: Derek Buitenhuis <derek.buitenhuis@gmail.com>
Signed-off-by: James Almer <jamrial@gmail.com>
---
 libavcodec/librav1e.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/libavcodec/librav1e.c b/libavcodec/librav1e.c
index f65d7841274..5052cac8960 100644
--- a/libavcodec/librav1e.c
+++ b/libavcodec/librav1e.c
@@ -508,12 +508,12 @@ static int librav1e_receive_packet(AVCodecContext *avctx, AVPacket *pkt)
         pkt->flags |= AV_PKT_FLAG_KEY;
 
     pkt->pts = pkt->dts = rpkt->input_frameno * avctx->ticks_per_frame;
+    rav1e_packet_unref(rpkt);
 
     if (avctx->flags & AV_CODEC_FLAG_GLOBAL_HEADER) {
         int ret = av_bsf_send_packet(ctx->bsf, pkt);
         if (ret < 0) {
             av_log(avctx, AV_LOG_ERROR, "extradata extraction send failed.\n");
-            rav1e_packet_unref(rpkt);
             av_packet_unref(pkt);
             return ret;
         }
@@ -521,7 +521,6 @@ static int librav1e_receive_packet(AVCodecContext *avctx, AVPacket *pkt)
         ret = av_bsf_receive_packet(ctx->bsf, pkt);
         if (ret < 0) {
             av_log(avctx, AV_LOG_ERROR, "extradata extraction receive failed.\n");
-            rav1e_packet_unref(rpkt);
             av_packet_unref(pkt);
             return ret;
         }
