diff -up ffmpeg-4.1/configure.force_dl ffmpeg-4.1/configure
--- ffmpeg-4.1/configure.force_dl	2019-01-08 18:29:28.359379710 +0100
+++ ffmpeg-4.1/configure	2019-01-08 18:30:45.598569521 +0100
@@ -3156,7 +3156,7 @@ libx262_encoder_deps="libx262"
 libx264_encoder_deps="libx264"
 libx264rgb_encoder_deps="libx264 x264_csp_bgr"
 libx264rgb_encoder_select="libx264_encoder"
-libx265_encoder_deps="libx265"
+libx265_encoder_deps="libx265 libdl"
 libxavs_encoder_deps="libxavs"
 libxavs2_encoder_deps="libxavs2"
 libxvid_encoder_deps="libxvid"
@@ -6042,7 +6042,7 @@ enabled bzlib && check_lib bzlib bzlib.h
 enabled  lzma && check_lib lzma   lzma.h lzma_version_number -llzma
 
 # On some systems dynamic loading requires no extra linker flags
-check_lib libdl dlfcn.h "dlopen dlsym" || check_lib libdl dlfcn.h "dlopen dlsym" -ldl
+check_lib libdl dlfcn.h "dlopen dlsym" -ldl
 
 check_lib libm math.h sin -lm
 
